hera_qm/ant_metrics.py

```diff
@@ -107,35 +167,54 @@ def mean_Vij_metrics(data, pols, antpols, ants, bls, xants=[], rawMetric=False):
 
 
 def compute_median_auto_power_dict(data, pols, reds):
-    '''Computes the median over frequency of the visibility squared, averaged over time.'''
+    """Compute the frequency median of the time averaged visibility squared.
+
+    Arguments:
+        data: Dictionary of visibility data.
+              keys are in the form (ant1, ant2, pol)
+        pols: List of polarizations to compute the median auto power.
+              allowed values ['xx', 'yy', 'xy', 'yx']
+        reds: List of lists of tuples of antenna numbers that make up redundant baseline groups.
+
+    Returns:
+        autoPower: Dictionary of the meidan of time average visibility squared.
+                   keys are in the form (ant1, ant2, pol)
+    """
     autoPower = {}
     for pol in pols:
         for bls in reds:
             for (i, j) in bls:
-                autoPower[i, j, pol] = np.median(np.mean(np.abs(data.get_data(i, j, pol))**2, axis=0))
+                tmp_power = np.abs(data[i, j, pol])**2
+                autoPower[i, j, pol] = np.median(np.mean(tmp_power, axis=0))
```

Comment:
good point, I will switch to nanmean/nanmedian. I'm not sure if we specifically need to be mindful of NaNs but it will help avert total failures if there is one.